### DISCUSSION TITLE
Which platform do we want to use for troubleshooting and discussing FIMS?

### DISCUSSION THREAD

**Discussion by Andrea-Havron-NOAA** [NOAA-FIMS/FIMS#472] [General]
I've sent out a survey asking for the team to vote on their preferred platform for a stackoverflow type discussion board for FIMS. Please use the survey form to mark your response and this thread to discuss pros/cons of different platforms, ask questions, or propose a different platform not listed. See the links below for more information about each platform.

The options are:
- Google Spaces ([link](https://support.google.com/a/users/answer/11219858?hl=en) )
- GitHub Discussions ([link](https://docs.github.com/en/discussions))
- Slack ([link](https://slack.com/))
- Discourse ([link](https://www.discourse.org/))

**Comment by Andrea-Havron-NOAA** [comment]
We should consider whether we want a public vs. private forum. I'm pretty sure Google Spaces would be NOAA internal. Slack requires an invitation to participate but channels can be public facing. GitHub Discussions require a github account to participate but is public facing. Discourse from my knowledge is open for anyone to post and is public facing.

**Reply by k-doering-NOAA** [reply]
actually, you can invite external folks to google spaces, although I'm assuming they do need a gmail?

**Reply by Andrea-Havron-NOAA** [reply]
Thanks for the update Kathryn!

**Comment by Andrea-Havron-NOAA** [comment]
Examples:

[FIMS Google Spaces](https://mail.google.com/mail/u/0/#chat/space/AAAAJqvKPpw) - We already have this platform!

[FIMS Discussion Board](https://github.com/orgs/NOAA-FIMS/discussions) - We already have this platform!

[Openscapes Slack](https://app.slack.com/client/THYEW0RC4/CHYEW203E) *This link will only work if you are also a member of Openscapes Slack

[The Stan Forums](https://discourse.mc-stan.org/)

**Comment by cmlegault** [comment]
Neither Discord nor Slack appear on the list of approved software at NEFSC currently. I do not know how difficult/possible it would be to get either approved.

**Reply by Andrea-Havron-NOAA** [reply]
You do not need to install Discorse (different from Discord, which is another possible option) or Slack on your computer to use the platforms. I have neither installed but have participated in both a Slack and Discorse group. Slack offers an "Open in Browser" option. With Discorse, you interact with the site like you would stackoverflow, you just need to login to the particular Discorse group and then you can post to the site using your browser. I expect the administrators would need the software installed, @ChristineStawitz-NOAA, am I correct about this?

**Reply by ChristineStawitz-NOAA** [reply]
Slack has been approved and is in use widely at the National Weather Service. Only the enterprise level is approved because it is able to keep conversations as a record, which is required. Andrea is correct that you can still use the web interface of the free version.

**Comment by Cole-Monnahan-NOAA** [comment]
Can we clarify the scope of this a bit? In my mind there are a few levels to consider. For example developers vs users, need quick help vs stats/algorithm question, and FIMS itself vs stock assessment best practices. Stan's discourse has many [categories](https://discourse.mc-stan.org/categories) which I think we would want to consider. I like that approach long-term. But it seems a bit formal for questions like Chris raised yesterday. I want to keep a space where we can ask those types of questions, particularly early in development as developers.

**Reply by k-doering-NOAA** [reply]
I agree with Cole!

**Reply by ChristineStawitz-NOAA** [reply]
I agree with wanting a different forum for internal vs external users. Although I don't think the internal forum need be private - for example, lots of development projects use Github as the development forum even though it is publicly available. Github discussions also has categories, which I like long term.

**Comment by kellijohnson-NOAA** [comment]
Thank you all for the great Discussions we have had in FIMS about which platforms to use. I am going to close this Discussion regarding where to discuss "troubleshooting and discussing" of FIMS because we do not currently have any plans to implement anything new in the immediate future. GitHub Discussions seems to be working well for things that need discussion and GitHub Issues for items that need changes to the code base. I hope to add more "channels" to the Discussions as we move further along in development.
